lib/Client/Transports/TransportHTTPPROTO.php

```diff
@@ -0,0 +1,59 @@
+<?php
+namespace LightStepBase\Client\Transports;
+
+class TransportHTTPPROTO {
+
+    protected $_host = '';
+    protected $_port = 0;
+    protected $_verbose = 0;
+
+    public function ensureConnection($options) {
+        $this->_verbose = $options['verbose'];
+
+        $this->_host = $options['collector_host'];
+        $this->_port = $options['collector_port'];
+
+        // The prefixed protocol is only needed for secure connections
+        if ($options['collector_secure'] == True) {
+            $this->_host = "ssl://" . $this->_host;
+        }
+    }
+
+    public function flushReport($auth, $report) {
+        if (is_null($auth) || is_null($report)) {
+            if ($this->_verbose > 0) {
+                error_log("Auth or report not set.");
+            }
+            return null;
+        }
+
+        $content = $report->toProto($auth)->serializeToString();
+
+        if ($this->_verbose >= 3) {
+            var_dump($content);
+        }
+
+        $header = "Host: " . $this->_host . "\r\n";
+        $header .= "User-Agent: LightStep-PHP\r\n";
+        $header .= "Accept: application/octet-stream\r\n";
+        $header .= "Content-Type: application/octet-stream\r\n";
+        $header .= "Content-Length: " . strlen($content) . "\r\n";
+        $header .= "Connection: keep-alive\r\n\r\n";
+
+        // Use a persistent connection when possible
+        $fp = @pfsockopen($this->_host, $this->_port, $errno, $errstr);
```

Comment:
I am by no means an expert in this, but this feels like a heavy weight way of sending a post request.
See this: https://stackoverflow.com/questions/5647461/how-do-i-send-a-post-request-with-php

Also, swallowing all of the errors with @ seems scary?

Reply:
i read those posts and tried other ways and this was the only one I could get working. it's the same way we currently do thrift. so how about i file an issue about the swallowing errors for follow-up later?

Reply:
https://github.com/lightstep/lightstep-tracer-php/issues/16